src/python/pants/backend/python/target_types_rules.py

```diff
@@ -181,12 +185,40 @@ def _classify_entry_points(
 async def resolve_python_distribution_entry_points(
     request: ResolvePythonDistributionEntryPointsRequest,
 ) -> ResolvedPythonDistributionEntryPoints:
-    field_value = request.entry_points_field.value
-    if field_value is None:
+    if request.entry_points_field:
+        address = request.entry_points_field.address
+        dist_field_value = request.entry_points_field.value
+        if dist_field_value is not None:
+            all_entry_points = cast(_EntryPointsDictType, dist_field_value)
+        else:
+            return ResolvedPythonDistributionEntryPoints()
```

Comment:
This might be more clear. Fits into "prefer early returns" from style guide: https://www.pantsbuild.org/docs/style-guide#prefer-early-returns-in-functions

```suggestion
    if request.entry_points_field:
        if request.entry_points_field.value is None:
            return ResolvedPythonDistributionEntryPoints()
        address = request.entry_points_field.address
        all_entry_points = cast(_EntryPointsDictType, request.entry_points_field.value)
```